From Nick.Holloway@pyrites.org.uk  Sun Feb  8 13:34:45 2009
From: Nick Holloway <Nick.Holloway@pyrites.org.uk>
Date: Sun, 25 Jan 2009 16:58:43 +0000
Subject: USB: Storage: Update unusual_devs entry for Datafab KECF-USB
Message-ID: <20090126185232.0504E3C0A2@alfie.demon.co.uk>

From: Nick Holloway <Nick.Holloway@pyrites.org.uk>

This device suffers from the off-by-one error when reporting the capacity,
so add US_FL_FIX_CAPACITY to the existing entry.

Signed-off-by: Nick Holloway <Nick.Holloway@pyrites.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/unusual_devs.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/storage/unusual_devs.h
+++ b/drivers/usb/storage/unusual_devs.h
@@ -1214,7 +1214,7 @@ UNUSUAL_DEV(  0x07c4, 0xa400, 0x0000, 0x
 		"Datafab",
 		"KECF-USB",
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
-		US_FL_FIX_INQUIRY ),
+		US_FL_FIX_INQUIRY | US_FL_FIX_CAPACITY ),
 
 /* Reported by Rauch Wolke <rauchwolke@gmx.net> */
 UNUSUAL_DEV(  0x07c4, 0xa4a5, 0x0000, 0xffff,
